src/proto/metrics2.proto

```diff
@@ -11,72 +11,137 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-syntax = "proto2";
+// This is copied and lightly edited from
+// github.com/prometheus/client_model/io/prometheus/client/metrics.proto
+// and finally converted to proto3 syntax to make it usable for the
```

Comment:
just out of curiosity, why not use the latest release. namely v2.48.1?  because you didn't want to include https://github.com/prometheus/prometheus/commit/644cb3be701fcfebbb639711d5cdc0c79946efd3 ?

Reply:
Not really. It's just that the PR took so much time to be released, and they kept updating the file. I'm not rushing to add more data on the messages if we don't have to, so it wasn't urgent to keep updating the metrics file if we do not support those extra features.

Reply:
i see. was asking because IIUC, the native histogram support in prometheus is still in a flux, so **might** want to use the latest proto definition. if the latest versions did not change the message struct since v2.44 , fine by me.

Reply:
This code is protected by a runtime flag, yes?

Reply:
you mean the `std::optional<>` ? yes, wanted to make sure we are using the latest schema used by the protected code, even if this piece of code is protected by a flag.

Reply:
@tchaikov, not the optional, the configuration flag that disables the protobuf code altogether